components/style/rule_tree/mod.rs

```diff
@@ -533,9 +533,9 @@ impl StrongRuleNode {
     }
 
     fn next_sibling(&self) -> Option<WeakRuleNode> {
-        // FIXME(emilio): Investigate what ordering can we achieve without
-        // messing things up.
-        let ptr = self.get().next_sibling.load(Ordering::SeqCst);
+        // Note that relaxed is fine because we take care of racing if needed
+        // with the correct memory ordering at the end of our search.
```

Comment:
Hm, what do you mean by "with the correct memory ordering" here? It seems like the reads below are relaxed as well.